src/modules/bulk-minter/components/change/ChangeDonationField.tsx

```diff
@@ -0,0 +1,280 @@
+          }}
+        >
+          {charity.name}
+        </Paragraph>
+        <FeatherIcon
+            className="creator-row-icon"
+            icon="external-link"
+            onClick={() => {window.open(CHANGE_BASE_URL + solana_address, "_blank")}}
+          />
+        <span style={{ marginLeft: 'auto' }}></span>
+        { enableSelectButton && <StyledSelectButton
+          onClick={onSelect}
+        >
+          {selectButtonText}
+        </StyledSelectButton> }
+        { !enableSelectButton && <AlreadyAddedNotice
+        >
+          {selectButtonText}
+        </AlreadyAddedNotice> }
+      </StyledCreatorsRow>  
+        <Paragraph
+          style={{ fontSize: 10, marginLeft:14, lineClamp: 4, textOverflow: 'ellipsis', overflow: 'hidden' }}
+        >
+          {description}
+        </Paragraph>
+        <Paragraph
+          style={{ fontSize: 10, marginLeft:14 }}
+          >
+          EIN: {ein}
+        </Paragraph>
+        </div>
+    );
+  };
+
+
+export const ChangeDonationField: FunctionComponent<Props> = ({
+    showDonationField,
+    creators,
+    updateDonationFieldState,
+    updateCreatorState,
+    updateShowErrorState
+  }: Props) => {
+    const [searchResults, setSearchResults] = useState<ChangeNonprofit[]>([]);
+    const [charityLoading, setLoading] = useState<boolean>(false);
+    const [isTyping, setIsTyping] = useState<boolean>(false);
+    const debounce = useMemo(() => new Debounce(), []);
+  
+    const addDonation = (nonProfit: ChangeNonprofit) => {
+        if (creators.length >= MAX_CREATOR_LIMIT) {
+          throw new Error('Max level of creators reached');
+        }
+        const existingCreators = creators.map((c) => c.address);
+        const indexOfDuplicate = existingCreators.findIndex((a) => nonProfit.crypto.solana_address === a);
+        if (indexOfDuplicate !== -1) {
+          throw new Error('All creator hashes must be unique');
+        }
+        const newShareSplit = 98 / (creators.length + 1);
+        updateCreatorState([
+          ...creators.map((c) => ({ ...c, share: newShareSplit })),
+          { address: nonProfit.crypto.solana_address, 
+            share: newShareSplit,
+            charityProps: {
+              isCharity: true,
+              displayName: nonProfit.name,
+              imageUrl:nonProfit.icon_url
+            } },
+        ]);
+        setSearchResults([]);
+        updateDonationFieldState(false);
+        updateShowErrorState(false);
+      };
+    const handleSearch = async (e: React.ChangeEvent<HTMLInputElement>) => {
+              setIsTyping(true);
+              debounce.debounceFcn(() => {
+                if (e.target.value === '') {
+                  setSearchResults([]);
+                  setLoading(false);
+                  setIsTyping(false);
+                }
+                else {performSearch(e.target.value)}
+              }, DEBOUNCE_INTERVAL_MS)
+          }
+          const performSearch = (textToSearch: string) => {
```

Comment:
IMO better to extract this to a well-named function within this file, since it is relatively self-contained and has little to do with markup